### PR TITLE
fix: should not setSchema('foo') in initialize when we set --schema=""

### PR DESCRIPTION
**Pull request by dacgray**
Hello!

* Type: bug fix
* issue:

#1162
This may be a better solution than #1255 

We expect that setting --schema="" will not add `setSchema()` in `initialize()`.

**Please check if fix provides desired behaviour.  If not I'll look into fixing #1255  ^^**

---

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines][:contrib:]
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR

---

- another pull request for this issue exists, but looks stale and this may be the better solution.
- this PR passes the option schema value from `AllModels::build` to `Models`.  
- we still use the schema from config to get our adapter instance
- this allows behaviour as described here: https://forum.phalconphp.com/discussion/15823/phalcon-tools-setschema-by-default-now

Thanks

[:contrib:]: https://github.com/phalcon/phalcon-devtools/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: scripts/Phalcon/Builder/Model.php [hunk 1]
```diff
@@ -159,7 +159,7 @@ public function build()
             $schema = Utils::resolveDbSchema($config->database);
         }
 
-        if ($schema) {
+        if (!empty($this->modelOptions->getOption('schema'))) {
```
Comment (Jeckerson) [inline scripts/Phalcon/Builder/Model.php]: There is same check on line 156

```
if ($this->modelOptions->hasOption('schema'))
```

Comment (dacgray) [reply scripts/Phalcon/Builder/Model.php]: `hasOption` uses isset.

The schema option is always set when coming from AllModels, so `$this->modelOptions->hasOption('schema')`, so we always setSchema in initialize.

If we want to follow the logic described here: https://forum.phalconphp.com/discussion/15823/phalcon-tools-setschema-by-default-now

We need to check if !empty.

If we don't want to follow the logic described in the forum post we need an init-schema flag.

**Comment by dacgray** [comment]
@Jeckerson 

What do you say?

- move forward with this solution 
- or fix https://github.com/phalcon/phalcon-devtools/pull/1255

**Comment by Jeckerson** [comment]
I'll check and tell you as soon as possible.

**Comment by Jeckerson** [comment]
@dacgray Can you create separate tests without altering current? To make sure that script executing that line `$schema = Utils::resolveDbSchema($config->database);`

**Comment by dacgray** [comment]
@Jeckerson @sergeyklay please take another look.

**Comment by dacgray** [comment]
@Jeckerson @sergeyklay good to merge?

**Comment by dacgray** [comment]
@Jeckerson @sergeyklay is there anything I need to do, or do you have a timeline for a 3.4.2 release?

**Comment by Jeckerson** [comment]
@dacgray https://github.com/phalcon/phalcon-devtools/milestone/4

**Pull request merged**